connectivity/check/logging.go

```diff
@@ -281,13 +281,21 @@ func (t *Test) Infof(format string, a ...interface{}) {
 	t.logf(info+" "+format, a...)
 }
 
+func (t *Test) failCommon() {
+	t.failed = true
+	t.flush()
+	if t.ctx.params.PauseOnFail {
+		t.log("Pausing after action failure, press the Enter key to continue:")
+		fmt.Scanln()
```

Comment:
Incidentally this is not great, because if you do ctrl+c it won't terminate the execution until you press enter to continue.

Reply:
Considering to add the following on top to solve this issue:

```
diff --git a/connectivity/check/logging.go b/connectivity/check/logging.go
index 230f0c196345..c1f18982f11f 100644
--- a/connectivity/check/logging.go
+++ b/connectivity/check/logging.go
@@ -4,9 +4,13 @@
 package check

 import (
+       "context"
        "fmt"
        "io"
+       "os"
+       "os/signal"
        "runtime"
+       "syscall"
        "time"
 )

@@ -286,7 +290,16 @@ func (t *Test) failCommon() {
        t.flush()
        if t.ctx.params.PauseOnFail {
                t.log("Pausing after action failure, press the Enter key to continue:")
-               fmt.Scanln()
+               cont := make(chan struct{})
+               go func() {
+                       fmt.Scanln()
+                       close(cont)
+               }()
+               ctx, _ := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+               select {
+               case <-cont:
+               case <-ctx.Done():
+               }
        }
 }

```

Reply:
magical